src/java.base/share/classes/sun/security/ssl/SSLLogger.java

```diff
@@ -74,40 +75,47 @@ public final class SSLLogger {
                 if (property.equals("help")) {
                     help();
                 }
-
                 logger = new SSLConsoleLogger("javax.net.ssl", p);
             }
             isOn = true;
+            // log almost everything for the "ssl" value.
+            // anything else specified with "ssl" value implies
+            // a subset of ssl debugging statements. The separator value
+            // for subset values has never been specified.
+            // Allow the expand operator here also (e.g. ssl:expand)
+            sslOn = property.equals("ssl") ||
+                    (property.startsWith("ssl")
+                            && property.length() == 10
+                            && !Character.isLetterOrDigit(property.charAt(3))
+                            && property.endsWith("expand"));
```

Comment:
With the current definition in the JSSE docs, `sslexpandhandshakeverbose` is an acceptable value, as are the tokens in any order.  e.g. `sslverbosehandshakeexpand` or `ssl|moohaha|handshake2354432verbose@#%^%SeanWasHere,keygen`.  This code here is position dependent and won't allow it.  

This approach needs a rethink.  Maybe an `EnumSet` initialized here that contains the active, known `ssl` options, that is then checked during the `isOn(checkPoints)`?  Just a thought.

Reply:
IMHO, the JSSE docs are pretty clear (even tho it's been broken for quite some time, sigh). 

> You do not have to have a separator between options, although a separator such as a colon (:) or a comma (,) helps readability. It does not matter what separators you use, and the ordering of the option keywords is also not important. 

No separators needed, no ordering enforced, could be something like this:

            EnumSet<Options> activeOptions = EnumSet.noneOf(Options.class);
            for (Options e : Options.values()) {
                if (property.contains(e.toString())) {
                    activeOptions.add(e);
                }
            }

and

        String[] options = checkPoints.split(",");
        for (String option : options) {
            option = option.trim().toLowerCase(Locale.ROOT);
            if (!activeOptions.contains(option.valueOf(option)) {
                return false;
            }
        }
        return true;

Reply:
The docs and implementation can be updated to a new specification if we choose to adopt a more sensible token parsing scheme.  Perhaps we can study that in a follow on issue.

 I've looked at the EnumSet option before also. IMO, it's yields maximum benefit if we start moving to a `"Debug.isOn(Option.SSL_HANDSHAKE)"` type call pattern in logging code in place of `"Debug.isOn("ssl, handshake")"`

I still see some issues with the current proposal (.e.g `"-Djavax.net.debug=sslctx"` would be an illegal option and yet it would turn on all ssl debug logs. Likewise `-Djavax.net.debug=typossl` has the odd behaviour of turning on all ssl debug code. It just strikes me as a broken system.

let me have a look at see if there' a hybrid solution that can be introduced to satisfy concerns.